Tools/clinic/clinic.py

```diff
@@ -4703,15 +4703,14 @@ def state_modulename_name(self, line: str | None) -> None:
                 function_name = fields.pop()
                 module, cls = self.clinic._module_and_class(fields)
 
-                for existing_function in (cls or module).functions:
-                    if existing_function.name == function_name:
-                        break
-                else:
-                    existing_function = None
-                if not existing_function:
-                    print("class", cls, "module", module, "existing", existing)
-                    print("cls. functions", cls.functions)
-                    fail("Couldn't find existing function " + repr(existing) + "!")
+                try:
+                    existing_function = next(
+                        f for f in (cls or module).functions if f.name == function_name
+                    )
+                except StopIteration:
```

Comment:
It looks too complicated. Would not the following code look more comprehensive?

```python
for f in (cls or module).functions:
    if f.name == function_name:
        existing_function = f
        break
else:
    ...  # fail
```

Reply:
Oh, it is actually the current code. So what is wrong with it?

Reply:
I found my refactor more readable (I wasn't a fan of the "implicit assignment" using the for loop), but I'll revert it since you disagree!

Reply:
d4b31b4